### DISCUSSION TITLE
kinet2pcb module issue

### DISCUSSION THREAD

**Discussion by Lakroft** [xesscorp/skidl#213] [Q&A]
Sorry for bothering again, this time I was trying to generate PCB directly from python code using generate_pcb() command mentioned in this book: https://shanemmattner.github.io/skidl_jupyter_book/chapter1/02_basic_usage.html
Warning message says kinet2pcb is missing though it already installed. When I try to import kinet2pcb directly into python shell it gives "No module named 'pcbnew'" error. 
I've found pcbnew.py in KiCad dir and added to PYTHONPATH, then error changed to " DLL load failed while importing _pcbnew: Module not found" and on this point I'm out of ideas how it can be fixed.

led.py code the same like in this discussion: https://github.com/devbisme/skidl/discussions/211 with only one change
```generate_pcb()```
instead of 
```generate_netlist()```

Issue present both on Windows and Mac machines.

Thank you.

![2024-06-10 23_28_15-Window](https://github.com/devbisme/skidl/assets/12113483/70cdb4a5-701e-4595-815e-eae7a4314adc)

**Comment by devbisme** [comment]
I'll take a look at this. I'll have to setup a Windows VM. It looks like you're running KiCad 6?

**Comment by devbisme** [comment]
I think the easiest thing to do is use the Python executable bundled with KiCad. The following is what I did for KiCad 6 running on a Windows 10 VM.

1. Prepend the KiCad Python executable to your path:
`set Path=C:\Program Files\KiCad\6.0\bin;%Path%`

2. Next, install SKiDL. The KiCad 6 executables don't include `pip`, but the `pip` module is included so use:
`python -m pip install skidl`

3. Next, modify the `led.py` program a bit to point to the directories where the symbols and footprints are:
```python
from skidl import *

# Tell SKiDL where the symbol libraries are.
lib_search_paths['kicad'].append('C:/Program Files/KiCad/6.0/share/kicad/symbols')

anodes = Bus('a', 6)
cathodes = Bus('k', 16)

led = Part('Device', 'D', footprint='Diode_SMD:D_0201_0603Metric', dest=TEMPLATE)

for a in anodes[1:4]:
    for k in cathodes[1:15]:
        led()['A', 'K'] += a, k

for i in range(2,6):
    for k in cathodes[i:i+10:5]:
        led()['A', 'K'] += anodes[5], k

ERC()

# Tell kinet2pcb where the footprint libraries are.
generate_pcb(fp_libs='C:/Program Files/KiCad/6.0/share/kicad/footprints')
```

4. Finally, run the program:
 `python led.py`

This should create a `led.kicad_pcb` file. I opened it with PCBNEW and it appeared to be correct.

**Reply by Lakroft** [reply]
Thanks, this worked, I've added a screenshot of generated pcb. 
Eventually I've created a separete alias `kipy` for kicad's python to keep my system's python as default. 

BTW looks like kicad's python pooled KICAD_SYMBOL_DIR so line `lib_search_paths['kicad'].append('C:/Program Files/KiCad/6.0/share/kicad/symbols')` is not nesessery, only the footprint. May be it also could be configured via env var?


![2024-06-15 23_45_53-Window](https://github.com/devbisme/skidl/assets/12113483/bb8e0ce8-0dfb-45e0-afb6-722b7026e77f)
![2024-06-16 00_14_19-Window](https://github.com/devbisme/skidl/assets/12113483/ffb3a400-1fa4-43ae-ae62-22b3e6514ec6)

**Comment by heeplr** [comment]
Had the same problem since I used venv which didn't include the PYTHONPATHs like my system python.
Solution was to create a `.pht` file inside my venv dir that contains the path to the directory where pcbnew is. 
In my case `env/lib/python3.12/site-packages/paths.pth` which contains
```
/usr/lib/python3.12/site-packages
```
